flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java

```diff
@@ -564,25 +529,6 @@ public int computeNumberOfPages(double fraction) {
 		return (int) (totalNumPages * fraction / numberOfSlots);
 	}
 
-	/**
-	 * Computes the memory size of the fraction per slot.
-	 *
-	 * @param fraction The fraction of the memory of the task slot.
-	 * @return The number of pages corresponding to the memory fraction.
-	 */
-	public long computeMemorySize(double fraction) {
```

Comment:
True, we could leave this method for now. I am a bit sceptical about these methods because they add another concern to this class which does not look like being directly related to the manager because it does not do anything with fractions. I would prefer to do it on the user side by calling `getMemorySize/NumberOfPages` etc methods or introduce a utility method to deduplicate the code.

My plan was that if we make memory manager per slot then the operator limit calculation is even more straightforward and I would refactor it afterwards.

Reply:
I think it makes sense to keep the two methods in memory manger, so that only memory manager but not operators needs to known the total managed memory size of the slot. Let's leave the discussion to when we make memory manager per slot.